securesystemslib/signer/_hsm_signer.py

```diff
@@ -192,15 +225,49 @@ def _find_key_values(
         return ECDomainParameters.load(bytes(params)), bytes(point)
 
     @classmethod
-    def import_(cls, hsm_keyid: Optional[int] = None) -> Tuple[str, SSlibKey]:
+    def _build_token_filter(cls) -> Dict[str, str]:
+        """Builds a token filter for the found slot/token.
+
+        The filter will include 'label' if one is found on token.
+
+        raises ValueError if less or more than 1 token/slot is found
+        """
+
+        lib = PYKCS11LIB()
+        slots: List[int] = lib.getSlotList(tokenPresent=True)
+        if len(slots) != 1:
+            raise ValueError(f"Expected 1 token/slot, found {len(slots)}")
+        filters = {}
+        tokeninfo = lib.getTokenInfo(slots[0])
+        # other possible fields include manufacturerID, model and serialNumber
+        for key in ["label"]:
+            try:
+                filters[key] = getattr(tokeninfo, key).strip()
+            except AttributeError:
+                pass
+
+        return filters
+
+    @classmethod
+    def import_(
+        cls,
+        hsm_keyid: Optional[int] = None,
+        token_filter: Optional[Dict[str, str]] = None,
+    ) -> Tuple[str, SSlibKey]:
         """Import public key and signer details from HSM.
 
+        Either only one token/slot must be present when importing or a
+        token_filter must be provided.
+
         Returns a private key URI (for Signer.from_priv_key_uri()) and a public
         key. import_() should be called once and the returned URI and public
         key should be stored for later use.
 
         Arguments:
             hsm_keyid: Key identifier on the token. Default is 2 (meaning PIV key slot 9c).
+            token_filter: Dictionary of token field names and values used to
+                filter the correct slot/token. If no filter is provided one is built
+                from the fields found on the token.
```

Comment:
> If no filter is provided one is built from the fields found on the token.

Why? Can't we just not filter, if no filter is provided?

Reply:
sure we could but my assumption is that filtering on "label" is the best default:
* it works fine if you have a single HW key (whether it has a label or not)
* when you buy a second HW key, things keep working and the signer URIS (that you have already created before) are still correctly identified if they have labels

If that assumption turns out false we can change the default to no filter.